src/compiler/emitter.ts

```diff
@@ -4502,7 +4516,195 @@ module ts {
+                                emitStart(accessors.setAccessor);
+                                if (member.flags & NodeFlags.Static) {
+                                    write("static ");
+                                }
+                                write("set ");
+                                emitMemberAccessForPropertyName((<MethodDeclaration>member).name);
+                                emitSignatureAndBody(accessors.setAccessor);
+                                emitEnd(accessors.setAccessor);
+                                emitTrailingComments(accessors.setAccessor);;
+                            }
+                        }
+                    }
+                });
+            }
+
+            function emitConstructorOfClass(node: ClassDeclaration, baseTypeNode: TypeReferenceNode) {
+                var saveTempCount = tempCount;
+                var saveTempVariables = tempVariables;
+                var saveTempParameters = tempParameters;
+                tempCount = 0;
+                tempVariables = undefined;
+                tempParameters = undefined;
+
+                var popFrame = enterNameScope();
+
+                // Emit the constructor overload pinned comments
+                forEach(node.members, member => {
+                    if (member.kind === SyntaxKind.Constructor && !(<ConstructorDeclaration>member).body) {
+                        emitPinnedOrTripleSlashComments(member);
+                    }
+                });
+
+                var ctor = getFirstConstructorWithBody(node);
+                if (ctor) {
+                    emitLeadingComments(ctor);
+                }
+                emitStart(<Node>ctor || node);
+
+                if (languageVersion < ScriptTarget.ES6) {
+                    write("function ");
+                    emitDeclarationName(node);
+                    emitSignatureParameters(ctor);
+                }
+                else {
+                    Debug.assert(languageVersion >= ScriptTarget.ES6, "Expected Script Target to be ES6 or above");
+                    write("constructor");
+                    if (ctor) {
+                        emitSignatureParameters(ctor);
+                    }
+                    else {
+                        // Based on EcmaScript6 section 14.5.14: Runtime Semantics: ClassDefinitionEvaluation.
+                        // If constructor is empty, then,
+                        //      If ClassHeritageopt is present, then
+                        //          Let constructor be the result of parsing the String "constructor(... args){ super (...args);}" using the syntactic grammar with the goal symbol MethodDefinition.
+                        //      Else,
+                        //          Let constructor be the result of parsing the String "constructor( ){ }" using the syntactic grammar with the goal symbol MethodDefinition
+                        if (baseTypeNode) {

```

Comment:
You are right. I was misreading the specification. It is say if the constructor is empty (i.e if we didn't emit the constructor), it will be evaluate as described. I agree that we should not emit the constructor if not specified